File: kotlin_jetpack_compose/app/src/main/java/com/example/testapp/data/supabase/repository/UserRepositoryImpl.kt
package com.example.testapp.data.supabase.repository

import com.example.testapp.data.supabase.model.mappers.korisnikISportasToDomain
import com.example.testapp.data.supabase.network.service.SupabaseUserService
import com.example.testapp.domain.model.SportasUser
import com.example.testapp.domain.repository.UserRepository
import io.github.jan.supabase.auth.auth

class UserRepositoryImpl(
    private val supabaseUserService: SupabaseUserService
) : UserRepository {

    override suspend fun getSportasById(id: String): SportasUser? {
        return try {
            val korisnik = supabaseUserService.getKorisnikById(id)
            val sportas = supabaseUserService.getSportasById(id)
            korisnikISportasToDomain(korisnik, sportas)
        } catch (e: Exception) {
            null
        }
    }

    override suspend fun getCurrentSportas(): SportasUser {
        return getSportasById(supabaseUserService.getCurrentUserId()!!)!!
    }
}

File: flutter/test_app/lib/data/supabase/repository/user_repository_impl.dart
import '../../../domain/model/sportas_user.dart';
import '../../../domain/repository/user_repository.dart';
import '../model/mapper/user_mapper.dart';
import '../network/service/user_service.dart';

class UserRepositoryImpl implements UserRepository {
  final UserService service;

  UserRepositoryImpl(this.service);

  @override
  Future<SportasUser?> getSportasUser(String id) async {
    try {
      final korisnik = await service.getKorisnik(id);
      final sportas = await service.getSportas(id);

      if (korisnik == null || sportas == null) return null;

      return mapToDomain(korisnik, sportas);
    } catch (_) {
      return null;
    }
  }
}